app/forms/fields.py

```diff
@@ -0,0 +1,149 @@
+import logging
+
+from wtforms import FormField, IntegerField, SelectField, SelectMultipleField, StringField, TextAreaField
+from wtforms import validators
+from wtforms.widgets import CheckboxInput, ListWidget, RadioInput, TextArea, TextInput
+
+from app.forms.date_form import get_date_form, get_month_year_form
+from app.validation.validators import IntegerCheck, PositiveIntegerCheck, NumberRange
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_field(answer, label, error_messages):
+    guidance = answer['guidance'] if 'guidance' in answer else ''
+
+    field = {
+        "Radio": get_select_field,
+        "Checkbox": get_select_field,
+        "Date": get_date_field,
+        "MonthYearDate": get_date_field,
+        "Currency": get_integer_field,
+        "Integer": get_integer_field,
+        "PositiveInteger": get_integer_field,
+        "Percentage": get_integer_field,
+        "TextArea": get_text_area_field,
+        "TextField": get_string_field,
+    }[answer['type']](answer, label, guidance, error_messages)
+
+    if field is None:
+        logger.info("Could not find field for answer type %s", answer['type'])
+
+    return field
+
+
+def build_choices(options):
+    choices = []
+    for option in options:
+        choices.append((option['value'], option['label']))
+    return choices
+
+
+def get_validators(answer, error_messages):
+    validate_with = [validators.optional()]
+
+    if answer['mandatory'] is True:
+        mandatory_message = error_messages['MANDATORY']
+
+        if 'validation' in answer and 'messages' in answer['validation'] \
+                and 'MANDATORY' in answer['validation']['messages']:
+            mandatory_message = answer['validation']['messages']['MANDATORY']
+
+        validate_with = [
+            validators.InputRequired(
+                message=mandatory_message,
+            ),
+        ]
+    return validate_with
+
+
+def get_string_field(answer, label, guidance, error_messages):
+    validate_with = get_validators(answer, error_messages)
+
+    return StringField(
+        label=label,
+        description=guidance,
+        widget=TextArea(),
```

Comment:
The reason I initially raised this is because I thought the wrong widget was being used as StringField should be a TextField if we were using it so I wasted time testing it and figuring out how it was still working to find out it's not being used at all